File: MainGame/FrontEndToolFS/SebastianPlanet/ShapeSettings.fs
namespace FrontEndToolFS.SebastianPlanet

type NoiseLayer() =
    member val enabled = true with get, set
    member val useFirstLayerAsMask = false with get, set
    member val noiseSettings = NoiseSettings()

type ShapeSettings() =
    member val planetRadius = 1f with get, set
    member val layerCount = 2 with get, set
    member val noiseLayers: NoiseLayer array = Array.init 2 (fun _ -> NoiseLayer()) with get, set

File: MainGame/game/SebastianPlanet/Settings/ShapeSettings.cs
using FrontEndToolFS.SebastianPlanet;
using Godot;

namespace ZeromaXPlayground.game.SebastianPlanet.Settings;

[Tool]
[GlobalClass]
public partial class ShapeSettings : Resource, IShapeSettings
{
    [Export] public float PlanetRadius { get; set; } = 1f;
    [Export] public NoiseLayer[] NoiseLayers { get; set; }
    public INoiseLayer[] noiseLayers => NoiseLayers; // 协变数组转换
}